File: lib/utils/official.dart
/// Identidad de la cuenta oficial de Talia (historias de sistema:
/// bienvenida y comunicados). NO existe como documento en `users`;
/// la UI hace special-casing por el flag `isOfficial` del Story.
///
/// Debe mantenerse sincronizado con la constante equivalente en
/// `functions/official.js` y con lo que escribe el backoffice.
class Official {
  Official._();

  /// userId constante del autor oficial.
  static const String userId = 'talia_official';

  /// Nombre mostrado.
  static const String userName = 'Talia';

  /// Asset usado como avatar oficial (ya cubierto por `assets/images/` en pubspec).
  static const String logoAsset = 'assets/images/logo.png';

  /// True si un userId corresponde a la cuenta oficial.
  static bool isOfficialUser(String? id) => id == userId;
}

File: functions/official.js
// ═══════════════════════════════════════════════════════════════
// CUENTA OFICIAL DE TALIA (historias de sistema)
// ═══════════════════════════════════════════════════════════════
//
// Identidad usada para historias de bienvenida y comunicados.
// NO existe como documento en `users`; el cliente hace special-casing
// por el flag `isOfficial` del Story.
//
// Debe mantenerse sincronizado con lib/utils/official.dart y con el
// backoffice que publica comunicados.

const OFFICIAL_USER_ID = "talia_official";
const OFFICIAL_USER_NAME = "Talia";

// Documento de config editable (desde el admin) para la historia de bienvenida.
const WELCOME_CONFIG_PATH = { collection: "app_config", doc: "welcome_story" };

module.exports = {
  OFFICIAL_USER_ID,
  OFFICIAL_USER_NAME,
  WELCOME_CONFIG_PATH,
};
